spanner/spannertest/funcs.go

```diff
@@ -103,3 +110,32 @@ var aggregateFuncs = map[string]aggregateFunc{
 		},
 	},
 }
+
+func evalMinMax(name string, isMin bool, values []interface{}, typ spansql.Type) (interface{}, spansql.Type, error) {
+	if typ.Array {
+		return nil, spansql.Type{}, fmt.Errorf("%s only supports non-array arguments, not %s", name, typ.SQL())
+	}
+	if len(values) == 0 {
+		// "Returns NULL if there are zero input rows".
+		return nil, typ, nil
+	}
+
+	// Compute running MIN/MAX.
+	// "Returns NULL if ... expression evaluates to NULL for all rows".
+	var minMax interface{}
```

Comment:
This won't work for `SELECT MIN(Name) FROM Staff` (or other `MIN(...)` expressions):
1. `minMax` will be nil here.
1. Line 135 will compare an actual non-nil value with the current value of `minMax`, which is nil.
1. `compareVals` will always return that a nil value is less than a non-nil value, which means that `minMax` will remain nil.

Reply:
Ah, good spot. Yeah, I meant to have an explicit minMax check. Fixed.